vm-setup/roles/v1a2_integration_test/tasks/verify.yml

```diff
@@ -0,0 +1,96 @@
+---
+  - name: Get the IP of controlplane node
+    shell: |
+      kubectl get bmh -n "{{ NAMESPACE }}" -o json | jq -r '.items[]
+      | select (.spec.consumerRef.name == "{{ CLUSTER_NAME }}-controlplane-0")
+      | .status.hardware.nics[] | select (.name == "eth1") | .ip'
+    register: controlplane_ip
+
+  - name: Define variable for controlplane node IP
+    set_fact:
+      bmh_ip: controlplane_ip.stdout
+
+  - name: Define username variable for ubuntu node
+    set_fact:
+      username: ubuntu
+    when: IMAGE_OS == "Ubuntu"
+
+  - name: Define username variable for target centos node
+    set_fact:
+      username: centos
+    when: IMAGE_OS == "Centos"
+
+  - name: Verification test
+    block:
+      - name: Waiting for controlplane to come up
+        wait_for:
+          port: 22
+          state: started
+          timeout: 1200
+          delay: 10
+
+      - name: Install jq
+        apt:
+          name: jq
+          state: present
+        when:
+          ansible_distribution == "Ubuntu"
+
+      - name: Install jq
+        yum:
+          name: jq
+          state: present
+        when:
+          ansible_distribution == "CentOS"
+
+      # Install Calico
+      - name: Download Calico manifest
+        get_url:
+          url: https://docs.projectcalico.org/v3.9/manifests/calico.yaml
+          dest: /tmp/
+        register: calico_manifest
+
+      - name: Replace the POD_CIDR in calico config
+        replace:
+          path: /tmp/calico.yaml
+          regexp: "192.168.0.0/16"
+          replace: '{{ baremetal_network }}'
+        register: updated_manifest
+
+      - name: Apply Calico manifest
+        k8s:
+          state: present
+          src: "/tmp/calico.yaml"
+          namespace: "{{ NAMESPACE }}"
+        register: install_cni
+
+      # Check for pods & nodes on the target cluster
+      - name: Check if k8s pods in running state
+        shell: |
+          kubectl get pods -A -o json | jq -r '.items[].status.phase'|
+          grep -v Running
+        environment:
+          KUBECONFIG: "{{ KUBECONFIG_PATH }}"
+        retries: 150
+        delay: 3
+        register: running_pods
+        failed_when: >
+          (running_pods.stderr != "") or
+          (running_pods.rc !=0) or
+          (running_pods.stdout != "")
+
+      - name: Check if k8s nodes in ready state
+        shell: "kubectl get nodes | grep -w Ready | wc -l"
+        environment:
+          KUBECONFIG: "{{ KUBECONFIG_PATH }}"
+        retries: 150
+        delay: 3
+        register: ready_nodes
+        failed_when: >
+          (ready_nodes.stderr != "") or
+          (ready_nodes.rc != 0) or
```

Comment:
there is some logic issue here, you are counting the number of lines that have Ready. Hopefully it would not be 0 so that pods are Ready.but the condition for failure is so that it would fail if there isn't 0 nodes ready. Maybe you should invert the grep, to count the lines that are not ready ?

Reply:
If I invert it to NotReady, then condition will fail only when status is NotReady?  But I was thinking that there might be something else other than NotReady that would indicate that nodes have failed or not ready. As such, if condition looks for NotReady there is a possibility that statuses like Unknown or Failed might be neglected?

Reply:
I meant using grep -v to invert the grep, not grepping "NotReady". Then you would catch anything that is not "ready". Anyways, I think it is ok as is. I don't see the problem anymore :p